Servers/WindowServer/WSWindowManager.cpp

```diff
@@ -703,46 +703,63 @@ void WSWindowManager::process_mouse_event(WSMouseEvent& event, WSWindow*& hovere
 
     WSWindow* event_window_with_frame = nullptr;
 
-    for_each_visible_window_from_front_to_back([&](WSWindow& window) {
-        auto window_frame_rect = window.frame().rect();
-        if (!window_frame_rect.contains(event.position()))
-            return IterationDecision::Continue;
-
-        if (&window != m_resize_candidate.ptr())
-            clear_resize_candidate();
-
-        // First check if we should initiate a drag or resize (Logo+LMB or Logo+RMB).
-        // In those cases, the event is swallowed by the window manager.
-        if (window.type() == WSWindowType::Normal) {
-            if (!window.is_fullscreen() && m_keyboard_modifiers == Mod_Logo && event.type() == WSEvent::MouseDown && event.button() == MouseButton::Left) {
-                hovered_window = &window;
-                start_window_drag(window, event);
-                return IterationDecision::Break;
+    if (m_active_input_window) {
+        // At this point, we have delivered the start of an input sequence to a
+        // client application. We must keep delivering to that client
+        // application until the input sequence is done.
+        //
+        // This prevents e.g. dragging on one window out of the bounds starting
+        // a drag in that other unrelated window, and other silly shennanigans.
+        auto translated_event = event.translated(-m_active_input_window->position());
+        deliver_mouse_event(*m_active_input_window, translated_event);
+        if (event.type() == WSEvent::MouseUp) {
```

Comment:
We need to take the mouse buttons into account here.
What if I press down both left & right, but then release only one of them?

Reply:
Hmm. I had assumed that there would only be a single "down" event, and that buttons can change from event to event, with a final 'up' event at the end when nothing else is pressed. Is that not the case? So we actually get

    Begin (left)
    Begin (right)
    Move
    Move
    End (left)
    Move
    End (right)

?

Reply:
That is correct. You can see how events are synthesized from mouse device input in `WSScreen::on_receive_mouse_data()`